File: solutions/0020_ValidParentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        open = ['(','{','[']
        closed = [')','}',']']
        stack = []
        for char in s:
            if char in closed:
                open_bracket = stack.pop() if stack else '#'
                if open_bracket != open[closed.index(char)]:
                    return False
            else:
                stack.append(char)
        
        if len(stack) == 0:
            return True
        else:
            return False

File: solutions/0020_ValidParentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    var stack = []
    var char_array = s.split("");
    var closed_brackets = [')','}',']'];
    var open_brackets = ['(','{','['];
    for(var i = 0; i<char_array.length;i++){
        if(closed_brackets.indexOf(char_array[i]) > -1){
            var open_bracket = open_brackets[closed_brackets.indexOf(char_array[i])]
            if(stack.length==0 || (open_bracket != stack.pop())) {
                return false;
            }
        } else {
            stack.push(char_array[i])
        }
    }
    if(stack.length == 0){
        return true
    } else {
        return false
    }
};